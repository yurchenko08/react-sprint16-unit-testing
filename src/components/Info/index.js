import { Component } from "react";
import getGitHubUser from "../../services/DataService";

class Info extends Component {

  constructor(props) {
    super(props);
    this.state = {};
  }

  async componentDidMount() {
    getGitHubUser(this.props.user)
      .then((response) => {
        this.setState(response.data);
      })
      .catch((error) => {
        this.setState({ error: "request error" });
      });
  }

  render() {
    return (
      <div>
        <h3>GitHub User Info</h3>
        <ul>
          {Object.keys(this.state).map((i) => (
            <li key={i}>
              {i}: {this.state[i]}
            </li>
          ))}
        </ul>
      </div>
    );
  }
}

export default Info;
