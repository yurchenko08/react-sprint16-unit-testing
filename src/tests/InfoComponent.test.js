import TestRenderer from 'react-test-renderer';
import Info from '../components/Info';
import React from 'react';

describe('unit tests for comnonent Info', () => {
  let componentInfo;

  beforeEach(() => {
    componentInfo = TestRenderer.create(<Info user='yurkovskiy' />);
  });

  it('Render corectly', () => {
    expect(componentInfo.toJSON()).toMatchSnapshot();
  });

  it('Test props', () => {
    const testInstance = componentInfo.toTree();
    expect(testInstance.props.user).toEqual('yurkovskiy');
  });
});
