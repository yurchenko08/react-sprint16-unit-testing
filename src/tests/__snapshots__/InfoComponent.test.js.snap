// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test comnonent Info Render corectly 1`] = `
<div>
  <h3>
    GitHub User Info
  </h3>
  <ul />
</div>
`;

exports[`unit tests for comnonent Info Render corectly 1`] = `
<div>
  <h3>
    GitHub User Info
  </h3>
  <ul />
</div>
`;
